lib/client/doc.js

```diff
@@ -426,14 +494,22 @@ function pushActionCallback(inflight, isDuplicate, callback) {
 //
 // Only one operation can be in-flight at a time. If an operation is already on
 // its way, or we're not currently connected, this method does nothing.
+//
+// If there are no pending ops, this method sends the pending presence data, if possible.
 Doc.prototype.flush = function() {
-  // Ignore if we can't send or we are already sending an op
-  if (!this.connection.canSend || this.inflightOp) return;
+  if (this.paused) return;
 
-  // Send first pending op unless paused
-  if (!this.paused && this.pendingOps.length) {
+  if (this.connection.canSend && !this.inflightOp && this.pendingOps.length) {
```

Comment:
The changes in logic changes here seem like minor cleanup unrelated to the presence feature. If so, these could be done in a separate PR, to more cleanly isolate the presence changes.